File: demo-frontend/src/types/customer.ts
export interface Customer {
    id: number;
    firstName: string;
    lastName: string;
    dateOfBirth: string;
    countryCode: string;
    countryName: string;
    phoneNumber: string;
    createdAt: string;
    updatedAt: string;
}

export interface CustomerRequest {
    firstName: string;
    lastName: string;
    dateOfBirth: string;
    countryCode: string;
    countryName: string;
    phoneNumber: string;
}

export interface PageResponse<T> {
    content: T[];
    page: {
        size: number;
        number: number;
        totalElements: number;
        totalPages: number;
    };
}

export interface ApiError {
    message: string;
    errors?: Record<string, string>;
}

File: demo-backend/src/main/kotlin/com/pratyush/demo_backend/entity/Customer.kt
package com.pratyush.demo_backend.entity

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.LocalDate
import java.time.LocalDateTime

@Entity
@Table(name = "customers")
data class Customer(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @Column(nullable = false, length = 50)
    val firstName: String,

    @Column(nullable = false, length = 50)
    val lastName: String,

    @Column(nullable = false)
    val dateOfBirth: LocalDate,

    @Column(nullable = false, length = 5)
    val countryCode: String,

    @Column(nullable = false, length = 100)
    val countryName: String,

    @Column(nullable = false, unique = true)
    val phoneNumber: String,

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    val createdAt: LocalDateTime = LocalDateTime.now(),

    @UpdateTimestamp
    @Column(nullable = false)
    val updatedAt: LocalDateTime = LocalDateTime.now()
)
